### ISSUE TITLE
Update to 0.4.4. no mailbox is loading any more

### ISSUE THREAD

**Issue by plastikschnitzer** [owncloud/mail#1479]
### Steps to reproduce
1. Update mail app to 0.4.4.
2. Open mail app

### Expected behaviour

App should work

### Actual behaviour

No mailbox content is loading, error message on top "error loading selected account". On the left side bar, the mailbox names and folders show up and on the right side, its loading and loading and loading … nothing.

### ownCloud Mail

**Number of accounts:** 5

#### Server configuration
**Operating system**: Gentoo Linux

**Web server:** Apache 2.4.10

**Database:** MySQL 5.6.19.

**PHP version:** 5.6.12.

**ownCloud version:** 9.0.1.

**Updated from an older ownCloud or fresh install:** updated

**List of activated apps:**

Activity
Announcement Center
Collaborative tags
Comments
Default encryption module
Deleted files
External sites
Federation
First run wizard
Mail Template Editor
Notifications
PDF Viewer
Provisioning API
Share Files
Text editor
Update notification
Versions
Video player
Calendar
Contacts
Mail
OwnNote
Piwik Tracking

##### ownCloud log (data/owncloud.log)

Exception: {"Exception":"Horde_Imap_Client_Exception","Message":"Authentication failed.","Code":102,"Trace":"#0 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(4554): Horde_Imap_Client_Socket->_responseCode(Object(Horde_Imap_Client_Interaction_Pipeline), Object(Horde_Imap_Client_Interaction_Server_Tagged))\n#1 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(4302): Horde_Imap_Client_Socket->_getLine(Object(Horde_Imap_Client_Interaction_Pipeline))\n#2 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(4230): Horde_Imap_Client_Socket->_sendCmdChunk(Object(Horde_Imap_Client_Interaction_Pipeline), Array)\n#3 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(846): Horde_Imap_Client_Socket->_sendCmd(Object(Horde_Imap_Client_Interaction_Pipeline))\n#4 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(512): Horde_Imap_Client_Socket->_tryLogin('PLAIN')\n#5 \/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Base.php(831): Horde_Imap_Client_Socket->_login(*** username and password replaced ***)\n#6 \/owncloud\/apps\/mail\/lib\/account.php(143): Horde_Imap_Client_Base->login(*** username and password replaced ***)\n#7 \/owncloud\/apps\/mail\/lib\/account.php(276): OCA\\Mail\\Account->getImapConnection()\n#8 \/owncloud\/apps\/mail\/lib\/account.php(322): OCA\\Mail\\Account->listMailboxes()\n#9 \/owncloud\/apps\/mail\/lib\/account.php(555): OCA\\Mail\\Account->getMailboxes()\n#10 \/owncloud\/apps\/mail\/lib\/account.php(433): OCA\\Mail\\Account->getSpecialFolder('inbox', false)\n#11 \/owncloud\/apps\/mail\/lib\/service\/unifiedaccount.php(89): OCA\\Mail\\Account->getInbox()\n#12 \/owncloud\/apps\/mail\/lib\/service\/unifiedaccount.php(68): OCA\\Mail\\Service\\UnifiedAccount->buildInbox()\n#13 \/owncloud\/apps\/mail\/lib\/controller\/folderscontroller.php(61): OCA\\Mail\\Service\\UnifiedAccount->getListArray()\n#14 [internal function]: OCA\\Mail\\Controller\\FoldersController->index(-1)\n#15 \/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(159): call_user_func_array(Array, Array)\n#16 \/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Mail\\Controller\\FoldersController), 'index')\n#17 \/owncloud\/lib\/private\/appframework\/app.php(110): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Mail\\Controller\\FoldersController), 'index')\n#18 \/owncloud\/lib\/private\/appframework\/routing\/routeactionhandler.php(45): OC\\AppFramework\\App::main('FoldersControll...', 'index', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#19 [internal function]: OC\\AppFramework\\routing\\RouteActionHandler->__invoke(Array)\n#20 \/owncloud\/lib\/private\/route\/router.php(276): call_user_func(Object(OC\\AppFramework\\routing\\RouteActionHandler), Array)\n#21 \/owncloud\/lib\/base.php(882): OC\\Route\\Router->match('\/apps\/mail\/acco...')\n#22 \/owncloud\/index.php(39): OC::handleRequest()\n#23 {main}","File":"\/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php","Line":5018}

**Comment by plastikschnitzer** [comment]
Interesting observation: When I do the update on my second server, I can not reproduce this. 
Its a FreeBSD Server with Apache 2.4.18 PHP 5.5.30 with mod_fcgid/2.3.9, same OC Version, same Apps as mentioned above + Document App and AntiVirus, no problems here.

Can someone reproduce this? 

How can I downgrade the app? I tried to replace the mail app folder with an older release but every time I tried to activate the app, ownCloud tells me "this app cannot be enabled because it makes the server unstable"

**Comment by ChristophWurst** [comment]
> Authentication failed

Did you change your password? :D

**Comment by plastikschnitzer** [comment]
No. 100 % not. Adding an account (in another ownCloud user account who has not mail accounts) is working, and the account is working too. Existing accounts always produce the error mentioned above.

**Comment by ChristophWurst** [comment]
Does deleting and re-adding the account help? Maybe the encrypted password cannot be decrypted. Which version did you use before? AFAIK we didn't touch ANY related code, the last releases we merely about polishing the front end.

**Comment by plastikschnitzer** [comment]
While the accounts are loading, deleting ends in an error message (account deletion failed) and adding a new account is not possible while the accounts are loading. I used version 0.4.2. before.

**Comment by ChristophWurst** [comment]
Do you have access to the database? Then you could delete the entry manually. Another way to debug this would be to attach a debugger to see what the decrypted password is. The latter one can also be checked by writing the password to the error_log.

**Comment by plastikschnitzer** [comment]
Database: Yes. Problem is I have many users and to convince them to set up new accounts is not the best idea …
I have no experience with debugger and decrypting passwords, sorry...

**Comment by ChristophWurst** [comment]
@owncloud/mail does anybody have an idea what might cause this error or how this could be debugged?

**Comment by plastikschnitzer** [comment]
I have an update: Deleting mail accounts and adding them again helps. I tried with some other user accounts and checked their OC-Mail App, and its working there, so strange. Also accounts with multiple mail accounts. The only difference to my first oc user account (which did not work) was, that the one which did not work, had a gmail account and some more imap accounts from other servers inside. But deleting and adding the gmail solved it. I am not really sure what happened here. :-/

**Comment by ChristophWurst** [comment]
Interesting, so this problem only occurs with gmail? There are some issues regarding failing account setup with gmail as this app is considered being a 'less secure app'.
Do you know if you changed the config while re-adding the account or it's the same as before? Did you use the auto config or manual config?

**Comment by plastikschnitzer** [comment]
… it seems like its just with gmail – but all the other accounts aside gmail were also not working. I deleted one account after another, 5 in total. As the gmail account was deleted, the others were not working. As I had no mail accounts inside the mail app any more, I set them all up again. Gmail with auto config without any problem, same way I did it like in my initial setup (I guess I changed something with my first setup in gmail, but I do not remember, it was about one year ago).
So in my eyes not a clear sign towards gmail as source of problems, but I did not see it in oc-accounts without gmail. All I have is this error log and my observation...
